src/MwApiPlugin.js

```diff
@@ -144,6 +170,16 @@ module.exports = {
 			async 'MwApi:CreateEntity'( { entityType, label, data } ) {
 				return createEntity( entityType, label, data );
 			},
+			async 'MwApi:CreateProperty'( { datatype, label, data } ) {
+				return createProperty( datatype, label, data );
+			},
+			async 'MwApi:GetOrCreatePropertyIdByDataType'( { datatype } ) {
```

Comment:
This is called `GetOrCreate`, but as far as I can see, if you create a new property type (one not already stored in `wikibasePropertyIds`), that create action would simply repeated when you do it for a second time.

Should the function also check in the wiki to see if a property exists for the datatype? Or should it not at least cache the properties it creates so that they're not created twice in the same session?

Reply:
In the old selenium tests it works like this, and only reuses a property if its set in an environment variable.
That said, I don't see a good reason not to cache it for the session.

> Should the function also check in the wiki to see if a property exists for the datatype? 
Since the property itself it arbitrary, finding any property with the right datatype should be fine for the tests. I like that idea better and will try to make it do that.

Reply:
The api currently doesn't support querying properties by datatype, which rules out looking for an existing property when an ID isn't specified. I did add caching though, so at most, 1 property is created per datatype and session.

Reply:
Sounds good - let's try that. Please rebase the version commit and I'll merge this.